File: my-folder/0053-maximum-subarray/solution.ts
function maxSubArray(nums: number[]): number {
    let maxSum = nums[0];

    let sum = nums[0];

    for (let i = 1; i < nums.length; i++) {
        if (nums[i] > sum && nums[i] + sum < nums[i]) {
            sum = nums[i];
        } else {
            sum += nums[i]
        }

        if (sum > maxSum) {
            maxSum = sum;
        }
    }

    return maxSum;
};

File: my-folder/0053-maximum-subarray/solution.py
class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        cur_sum = largest_sum = float('-inf')

        for num in nums:
            cur_sum = max(cur_sum + num, num)
            largest_sum = max(largest_sum, cur_sum)
        
        return largest_sum
